content/blog/persisting-kubernetes-workloads-with-amazon-efscsi-volumes-using-pulumi-sdks/index.md

```diff
@@ -0,0 +1,397 @@
+-rw-------    1 nishidavidson  staff     21 Jun 18 18:22 .gitignore
+-rw-r--r--    1 nishidavidson  staff     32 Jun 18 18:22 Pulumi.dev.yaml
+-rw-------    1 nishidavidson  staff     91 Jun 18 18:22 Pulumi.yaml
+-rw-------    1 nishidavidson  staff    273 Jun 18 18:22 index.ts
+drwxr-xr-x   95 nishidavidson  staff   3040 Jun 18 18:22 node_modules
+-rw-r--r--    1 nishidavidson  staff  50650 Jun 18 18:22 package-lock.json
+-rw-------    1 nishidavidson  staff    228 Jun 18 18:22 package.json
+-rw-------    1 nishidavidson  staff    522 Jun 18 18:22 tsconfig.json
+
+```
+
+## Step 2: Create an EKS cluster, an EFS endpoint and mount targets in the public subnets of the EKS cluster:
+
+The code below can be pasted in `index.ts` to create a default EKS cluster with two public subnets in the new VPC. We then declare EFS endpoint and mount the same to both subnets so the the EKS worker nodes can access the filesystem. 
+
+```typescript
+import * as aws from "@pulumi/aws";
+import * as awsx from "@pulumi/awsx";
+import * as eks from "@pulumi/eks";
+import * as k8s from "@pulumi/kubernetes";
+
+//* STEP 1: Create an EKS cluster, an EFS endpoint and mount the EFS endpoint in each public subnet of the EKS cluster
+
+// Create an EKS Cluster
+const vpc = new awsx.Network("vpc", { usePrivateSubnets: false });
+export const cluster = new eks.Cluster("eks-cluster", {
+  vpcId             : vpc.vpcId,
+  subnetIds         : vpc.publicSubnetIds,
+  instanceType      : "t2.medium",
+  version           : "1.12",
+  nodeRootVolumeSize: 200,
+  desiredCapacity   : 2,
+  maxSize           : 3,
+  minSize           : 2,
+  deployDashboard   : true,
+  vpcCniOptions     : {
+    warmIpTarget    : 4,
+  },
+});
+
+const nodesg = cluster.nodeSecurityGroup;
+export const nodesubnetId = cluster.core.subnetIds;
+export const clusterName = cluster.eksCluster.name;
+export const kubeconfig = cluster.kubeconfig.apply(JSON.stringify);
+
+// Create an EFS endpoint 
+export const efsFilesystem = new aws.efs.FileSystem("MyEFS", {
+  tags: { Name: "myEFS" },
+});
+
+export const efsFilesystemId = efsFilesystem.id;
+
+// Create a mounttarget in each of the EKS cluster VPC's AZs so that EC2 instances across the VPC can access the filesystem
+new aws.efs.MountTarget("MyEFS-MountTarget1", {
+  fileSystemId: efsFilesystemId,
+  securityGroups: [ nodesg.id ],
+  subnetId: nodesubnetId[0],
+});
+
+new aws.efs.MountTarget("MyEFS-MountTarget2", {
+  fileSystemId: efsFilesystemId,
+  securityGroups: [ nodesg.id ],
+  subnetId: nodesubnetId[1],
+});
+
+```
+
+## Step 3: Create the CSI driver node and controller components
+
+Currently, only static provisioning for AWS EFS CSI drivers is supported by SIG-AWS. Implies, Amazon EFS filesystem needs to be created manually first. After that it can be mounted inside a container as a volume using the driver. The code below will allow you to deploy allow the CSI Driver components on the Amazon EKS cluster.
```

Comment:
```suggestion
Currently, only static provisioning for AWS EFS CSI drivers is supported by SIG-AWS, which implies the Amazon EFS filesystem needs to be created manually first.  After creation, it can be mounted inside a container as a volume using the driver. The code below will allow you to deploy allow the CSI Driver components on the Amazon EKS cluster.
```

Reply:
absorbed this suggestion and made an additional edit.